### PR TITLE
MNT: Silence internal command calls' result rendering in add-readme

### PR DESCRIPTION
**Pull request by adswa**
This updates `add-readme`  according to #6461. Previously, 'datalad add-readme' resulted in double reporting that looked
like this:
```
datalad add-readme
[WARNING] Found no aggregated metadata info file /tmp/abcd/more/.datalad/metadata/aggregate_v1.json. You will likely need to either update the dataset from its original location or reaggregate metadata locally.
metadata(impossible): . (dataset) [Dataset at . contains no aggregated metadata on this path]
[WARNING] Could not obtain dataset metadata, proceeding without
add_readme(ok): /tmp/abcd/more/README.md (file)
add(ok): README.md (file)
save(ok): . (dataset)
action summary:
  add (ok: 1)
  save (ok: 1)
add(ok): README.md (file)
save(ok): . (dataset)
action summary:
  add (ok: 1)
  add_readme (ok: 1)
  save (ok: 1)
```
With this change applied, it looks like this:
```
datalad add-readme
[WARNING] Found no aggregated metadata info file /tmp/abcd/more/more/more/.datalad/metadata/aggregate_v1.json. You will likely need to either update the dataset from its original location or reaggregate metadata locally.
[WARNING] Could not obtain dataset metadata, proceeding without
add_readme(ok): /tmp/abcd/more/more/more/README.md (file)
add(ok): README.md (file)
save(ok): . (dataset)
action summary:
  add (ok: 1)
  add_readme (ok: 1)
  save (ok: 1)
```

### CODE REVIEW

**Review by christian-monch — COMMENTED** [review]
There might be an unlock in line 91 the should be changed as well

File: datalad/local/add_readme.py [hunk 1]
```diff
@@ -101,12 +101,17 @@ def __call__(filename='README.md',
                 dataset.save(
                     path='.gitattributes',
                     message="[DATALAD] Configure README to be in Git",
-                    to_git=True
+                    to_git=True,
+                    return_type='generator',
+                    result_renderer='disabled'
```
Comment (mih) [inline datalad/local/add_readme.py:106]: needs to be wrapped into `yield from` or would stop doing anything

File: datalad/local/add_readme.py [hunk 2]
```diff
@@ -236,5 +241,8 @@ def __call__(filename='README.md',
                 fpath,
                 message='[DATALAD] added README',
                 result_filter=None,
-                result_xfm=None):
+                result_xfm=None,
+                return_type='generator',
+                result_renderer='disabled'
+        ):
```
Comment (mih) [inline datalad/local/add_readme.py:247]: The whole call could be reduced to `yield from` instead of the `for` loop

**Review by mih — CHANGES_REQUESTED** [review]
I made a few remarks, one points to a bug. Thx!

File: datalad/local/add_readme.py (hunk 1 above)
Comment (adswa) [reply datalad/local/add_readme.py:106]: oh thanks for catching this! it is a bit worrisome that the tests did not pick this up...

File: datalad/local/add_readme.py (hunk 2 above)
Comment (adswa) [reply datalad/local/add_readme.py:247]: thanks, done!

**Review by mih — CHANGES_REQUESTED** [review]
LGTM. But as @christian-monch mentioned, there is an unlock call on line 91. It should likely yield is results with rendering disabled.

**Comment by adswa** [comment]
Oh sorry, I did not see this comment!

**Comment by yarikoptic** [comment]
I didn't spot anything else, looks LGTM, let's proceed

**Pull request merged**